service_configuration_lib/spark_config.py

```diff
@@ -252,22 +249,17 @@ def _append_sql_shuffle_partitions_conf(spark_opts: Dict[str, str]) -> Dict[str,
         str(spark_opts['spark.dynamicAllocation.maxExecutors']) != 'infinity'
     ):
 
-        num_partitions_dra = 2 * (
+        num_partitions_dra = 3 * (
             int(spark_opts.get('spark.dynamicAllocation.maxExecutors', 0)) *
             int(spark_opts.get('spark.executor.cores', DEFAULT_EXECUTOR_CORES))
         )
         num_partitions = max(num_partitions, num_partitions_dra)
 
     num_partitions = num_partitions or DEFAULT_SQL_SHUFFLE_PARTITIONS
+    _append_spark_config(spark_opts, 'spark.sql.shuffle.partitions', str(num_partitions))
+    _append_spark_config(spark_opts, 'spark.sql.files.minPartitionNum', str(num_partitions))
```

Comment:
I am not particularly sure if we should set this config by default.
I think this should be a lower value. Very low partition sizes is good for data-pipeline publishing but not good bunch of other actions, especially would probably take up very less executor resource, leading to higher cost.

I still incline towards `repartition` logic.

Side-note: We should still remove this, as it's defaults to `spark.default.parallelesim`
![image](https://user-images.githubusercontent.com/7134353/203157663-3fb614b0-68bb-40bf-bde5-91df25044b7a.png)